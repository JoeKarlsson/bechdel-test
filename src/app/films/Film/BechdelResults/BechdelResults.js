import React from 'react';
import PropTypes from 'prop-types';

const BechdelResults = (props) => {

	const {
		bechdelScore,
		numScenesPass,
		numScenesDontPass,
		numOfFemalesChars,
		numOfMaleChars,
		numOfFemalesCharsWithDialogue,
		numOfMaleCharsWithDialogue,
		totalLinesFemaleDialogue,
		totalLinesMaleDialogue,
	} = props.bechdelResults;

	return (
		<div className="BechdelResults">
			<span className="catName">Bechdel Score:</span>{' '}
			{bechdelScore} of 3<br />
			<span className="catName">
			Number of Scenes that pass:
			</span>{' '}
			{numScenesPass}
			<br />
			<span className="catName">
			Number of Scenes that dont pass:
			</span>{' '}
			{numScenesDontPass}
			<br />
			<span className="catName">
			Number Of Females Characters:
			</span>{' '}
			{numOfFemalesChars}
			<br />
			<span className="catName">
			Number Of Male Characters:
			</span>{' '}
			{numOfMaleChars}
			<br />
			<span className="catName">
			Number of Females Characters With Dialogue:
			</span>{' '}
			{numOfFemalesCharsWithDialogue}
			<br />
			<span className="catName">
			Number of Male Characters With Dialogue:
			</span>{' '}
			{numOfMaleCharsWithDialogue}
			<br />
			<span className="catName">
			Total Lines of Female Dialogue:
			</span>{' '}
			{totalLinesFemaleDialogue}
			<br />
			<span className="catName">
			Total Lines of Male Dialogue:
			</span>{' '}
			{totalLinesMaleDialogue}
			<br />
		</div>
	);
};

BechdelResults.propTypes = {
	bechdelResults: PropTypes.shape({
		bechdelScore: PropTypes.number,
		numScenesPass: PropTypes.number,
		numScenesDontPass: PropTypes.number,
		numOfFemalesChars: PropTypes.number,
		numOfMaleChars: PropTypes.number,
		numOfFemalesCharsWithDialogue: PropTypes.number,
		numOfMaleCharsWithDialogue: PropTypes.number,
		totalLinesFemaleDialogue: PropTypes.number,
		totalLinesMaleDialogue: PropTypes.number,
	}),
};

BechdelResults.defaultProps = {
	bechdelResults: {
		bechdelScore: 0,
		numScenesPass: 0,
		numScenesDontPass: 0,
		numOfFemalesChars: 0,
		numOfMaleChars: 0,
		numOfFemalesCharsWithDialogue: 0,
		numOfMaleCharsWithDialogue: 0,
		totalLinesFemaleDialogue: 0,
		totalLinesMaleDialogue: 0,
	},
};

export default BechdelResults;
