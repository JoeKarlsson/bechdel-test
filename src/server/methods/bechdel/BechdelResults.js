class BechdelResults {
	constructor() {
		this._pass = false;
		this._bechdelScore = 0;
		this._numScenesPass = 0;
		this._numScenesDontPass = 0;
		this._numOfFemalesChars = 0;
		this._numOfMaleChars = 0;
		this._numOfFemalesCharsWithDialogue = 0;
		this._numOfMaleCharsWithDialogue = 0;
		this._totalLinesFemaleDialogue = 0;
		this._totalLinesMaleDialogue = 0;
		this._characters = [];
		this._scenesThatPass = [];
		this._scenes = [];
	}

	get bechdelPass() {
		return this._pass;
	}

	set bechdelPass(passed) {
		this._pass = passed;
		return this._pass;
	}

	get bechdelScore() {
		return this._bechdelScore;
	}

	set bechdelScore(score) {
		if (score >= 3) {
			this._bechdelScore = 3;
		}
		if (this._bechdelScore === 3) {
			this._pass = true;
			return this._bechdelScore;
		}
		this._bechdelScore = score;
		return this._bechdelScore;
	}

	get numScenesPass() {
		return this._numScenesPass;
	}

	numScenesPassIncrement() {
		return this._numScenesPass++;
	}

	get numScenesDontPass() {
		return this._numScenesDontPass;
	}

	numScenesDontPassIncrement() {
		return this._numScenesDontPass++;
	}

	get numOfFemalesChars() {
		return this._numOfFemalesChars;
	}

	numOfFemalesCharsIncrement() {
		return this._numOfFemalesChars++;
	}

	get numOfMaleChars() {
		return this._numOfMaleChars;
	}

	numOfMaleCharsIncrement() {
		return this._numOfMaleChars++;
	}

	get numOfFemalesCharsWithDialogue() {
		return this._numOfFemalesCharsWithDialogue;
	}

	numOfFemalesCharsWithDialogueIncrement() {
		return this._numOfFemalesCharsWithDialogue++;
	}

	get numOfMaleCharsWithDialogue() {
		return this._numOfMaleCharsWithDialogue;
	}

	numOfMaleCharsWithDialogueIncrement() {
		return this._numOfMaleCharsWithDialogue++;
	}

	get totalLinesFemaleDialogue() {
		return this._totalLinesFemaleDialogue;
	}

	totalLinesFemaleDialogueIncrement() {
		return this._totalLinesFemaleDialogue++;
	}

	totalLinesFemaleDialogueAdd(count) {
		this._totalLinesFemaleDialogue += count;
		return this._totalLinesFemaleDialogue;
	}

	get totalLinesMaleDialogue() {
		return this._totalLinesMaleDialogue;
	}

	totalLinesMaleDialogueIncrement() {
		return this._totalLinesMaleDialogue++;
	}

	totalLinesMaleDialogueAdd(count) {
		this._totalLinesMaleDialogue += count;
		return this._totalLinesMaleDialogue;
	}

	get scenesThatPassBechdel() {
		return this._scenesThatPass;
	}

	addBechdelPassingScene(scene) {
		this._scenesThatPass.push(scene);
		return this._scenesThatPass;
	}

	get characters() {
		return this._characters;
	}

	set characters(characters) {
		this._characters = characters;
		return this._characters;
	}

	addCharacter(character) {
		this._characters.push(character);
		return this._characters;
	}

	get scenes() {
		return this._scenes;
	}

	addScene(scene) {
		this._scenes.push(scene);
		return this._scenes;
	}

	getBechdelResults() {
		return {
			pass: this._pass,
			bechdelScore: this._bechdelScore,
			numScenesPass: this._numScenesPass,
			numScenesDontPass: this._numScenesDontPass,
			numOfFemalesChars: this._numOfFemalesChars,
			numOfMaleChars: this._numOfMaleChars,
			numOfFemalesCharsWithDialogue: this._numOfFemalesCharsWithDialogue,
			numOfMaleCharsWithDialogue: this._numOfMaleCharsWithDialogue,
			totalLinesFemaleDialogue: this._totalLinesFemaleDialogue,
			totalLinesMaleDialogue: this._totalLinesMaleDialogue,
			characters: this._characters,
			scenesThatPass: this._scenesThatPass,
			// scenes: this._scenes,
		};
	}

	reset() {
		this._pass = false;
		this._bechdelScore = 0;
		this._numScenesPass = 0;
		this._numScenesDontPass = 0;
		this._numOfFemalesChars = 0;
		this._numOfMaleChars = 0;
		this._numOfFemalesCharsWithDialogue = 0;
		this._numOfMaleCharsWithDialogue = 0;
		this._totalLinesFemaleDialogue = 0;
		this._totalLinesMaleDialogue = 0;
		this._characters = [];
		this._scenesThatPass = [];
		this._scenes = [];

		return true;
	}
}

const bechdelResults = new BechdelResults();

module.exports = bechdelResults;
