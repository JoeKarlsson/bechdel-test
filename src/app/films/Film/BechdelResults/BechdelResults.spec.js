import React from 'react';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BechdelResults from './BechdelResults';

configure({ adapter: new Adapter() });

describe('BechdelResults', () => {
	let wrapper;

	global.requestAnimationFrame = (callback) => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<BechdelResults />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<BechdelResults />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should be selectable by the class `BechdelResults`', () => {
				expect(wrapper.is('.BechdelResults')).toBe(true);
			});

			it('should mount in the full DOM', () => {
				expect(wrapper.find('.BechdelResults').length).toBe(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
