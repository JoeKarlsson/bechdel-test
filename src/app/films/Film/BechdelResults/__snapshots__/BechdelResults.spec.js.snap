// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BechdelResults rendering initial state is rendered 1`] = `
<div
  className="BechdelResults"
>
  <span
    className="catName"
  >
    Bechdel Score:
  </span>
   
  0
   of 3
  <br />
  <span
    className="catName"
  >
    Number of Scenes that pass:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Number of Scenes that dont pass:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Number Of Females Characters:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Number Of Male Characters:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Number of Females Characters With Dialogue:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Number of Male Characters With Dialogue:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Total Lines of Female Dialogue:
  </span>
   
  0
  <br />
  <span
    className="catName"
  >
    Total Lines of Male Dialogue:
  </span>
   
  0
  <br />
</div>
`;
